Naughts_and_Crosses.py

```diff
@@ -1,122 +1,237 @@
-board_state_start = [[' ',' ',' '],[' ',' ',' '],[' ',' ',' ']]
-win_condition = ()
-board_state = board_state_start
-s = None
-
-win_condition = [
-	#Horizontal Wins
-	[board_state[0][0], board_state[0][1], board_state[0][2]],
-	[board_state[1][0], board_state[1][1], board_state[1][2]],
-	[board_state[2][0], board_state[2][1], board_state[2][2]],
-
-	#Vertical Wins
-	[board_state[0][0], board_state[1][0], board_state[2][0]],
-	[board_state[0][1], board_state[1][1], board_state[2][1]],
-	[board_state[0][2], board_state[1][2], board_state[2][2]],
-
-	#TR BL
-	[board_state[0][0], board_state[1][1], board_state[2][2]],
-
-	#TL BR
-	[board_state[0][2], board_state[1][1], board_state[2][0]]
-]
+import random
+Playing = True
+
+class Board():
+    def __init__(self): #Finds the board size, generates the empty board, win condition list and empty spaces list
+        self._board_size()
+        self.board_state = [[' ' for i in range(self.board_size)] for i in range(self.board_size)]
+        self.generate_win_condition()
+        self.generate_empty_spaces()
+    
+    def generate_empty_spaces(self):  #Generate the empty spaces from the board size
+        self.empty_spaces = []
+        for i in range(self.board_size):
+            for j in range(self.board_size):
+                self.empty_spaces.append([i,j])
+
+    def generate_win_condition(self): #Generates the win condition set from the board size
+        self.win_condition = []
+
+        #Vertical Wins
+        for i in range(self.board_size):
+            _sub_win_condition = []
+            for j in range(self.board_size):
+                _sub_win_condition.append([i, j])
```

Comment:
No underscore, you are defining it multiple times, I'll explain at another time.